controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -114,6 +114,18 @@ func AkoAddonSecretDataYaml(cluster *clusterv1.Cluster, obj *akoov1alpha1.AKODep
 	if err != nil {
 		return "", err
 	}
+
+	//Avoid setting DeleteConfig for management cluster
+	if cluster.Namespace != akoov1alpha1.TKGSystemNamespace {
+		if deleteConfig, exists := cluster.Labels[akoov1alpha1.AviClusterDeleteConfigLabel]; exists {
+			if deleteConfig == "true" {
+				secret.LoadBalancerAndIngressService.Config.AKOSettings.DeleteConfig = "true"
```

Comment:
maybe add a integration test case under `akodeploymentconfig_controller_intg_test.go` to cover this path?

Reply:
I have added a unit-test to cover this path. Anyways, I will explore adding an integration test

Reply:
Added a integration test case